wgctrl-rs/src/device.rs

```diff
@@ -103,14 +109,137 @@ pub struct DeviceInfo {
     pub(crate) __cant_construct_me: (),
 }
 
+type RawInterfaceName = [c_char; libc::IFNAMSIZ];
+
+/// The name of a Wireguard interface device.
+#[derive(PartialEq, Eq, Clone, Copy)]
+pub struct InterfaceName(RawInterfaceName);
+
+impl TryFrom<&str> for InterfaceName {
```

Comment:
I'm very sorry for suggesting `TryFrom<&str>` here, I think [`str::FromStr`](https://doc.rust-lang.org/std/str/trait.FromStr.html) is more flexible, since it can be easily used in `structopt`, for example. Thankfully it's basically the same signature. My bad.

Reply:
No problem, I moved it to use `FromStr` instead.